substrate/frame/support/src/storage/storage_noop_guard.rs

```diff
@@ -37,11 +37,40 @@
 /// });
 /// ```
 #[must_use]
-pub struct StorageNoopGuard(sp_std::vec::Vec<u8>);
+pub struct StorageNoopGuard {
+	storage_root: sp_std::vec::Vec<u8>,
+	error_message: &'static str,
```

Comment:
Why does it need to be static? `String` would allow it to be used with dynamic messages as well.

Reply:
I used the static lifetime because of how it is used in `Drop`. happy to switch to String though, I hadn't realised it was just for testing